index/scorch/persister.go

```diff
@@ -871,30 +888,118 @@ func (s *Scorch) removeOldData() {
 // rollback'ability.
 var NumSnapshotsToKeep = 1
 
+// RollbackSamplingInterval controls how far back we are looking
+// in the history to get the rollback points.
+// For example, the default value of 10 minutes ensures that the
+// protected snapshots (NumSnapshotsToKeep = 3) are:
+//    the very latest snapshot(ie the current one),
+//    the snapshot that was persisted 10 minutes before the current one,
+//	  the snapshot that was persisted 20 minutes before the current one
+var RollbackSamplingInterval = 2 * time.Minute
+
+// Controls what portion of the earlier rollback points to retain during
+// a infrequent/sparse mutation scenario
+var RollbackRetentionFactor = float64(0.5)
```

Comment:
Not sure I understand the purpose of this.

Reply:
My bad, I've updated the description which clarifies this part